Add tests for SplitContainer

kitbdev opened the pull request:
Adds tests for SplitContainer.
Including dragger visibility, mouse cursor, collapsed state, as well as default position, setting split offset, and dragging in various child expand flag configurations. Let me know if there are better ways to organize it or other important tests to include. 

The `check_position` helper functions could be simplified to work with only two children, but since I'm working on adding multiple children and want to add tests for that, I figured I'll leave it like this.

When testing collapsed behavior, see the updated description as it is inaccurate: https://github.com/godotengine/godot/pull/97373

kitbdev committed:
Add tests for SplitContainer

AThousandShips commented on doc/classes/SplitContainer.xml:
```diff
@@ -30,7 +30,7 @@
 	<members>
 		<member name="collapsed" type="bool" setter="set_collapsed" getter="is_collapsed" default="false">
-			If [code]true[/code], the area of the first [Control] will be collapsed and the dragger will be disabled.
+			If [code]true[/code], the dragger will be disabled and the children will be sized as if the [member split_offset] was [code]0[/code].
```
Unrelated, please make a separate commit so this can be cherry picked (we don't cherry pick tests)

kitbdev replied:
- In that case, I should do the same with my https://github.com/godotengine/godot/pull/97255

Mickeon commented:
Would be nice to have this merged sooner than later, still waiting for Test review and potentially a rebase.

akien-mga reviewed (approved):
Having difficulty getting this reviewed, but it's a low risk test and kitbdev is pretty well versed in the Godot API, so YOLO.

I'd suggest rebasing first to make sure it still builds and passes checks.

kitbdev commented on tests/scene/test_split_container.h:
```diff
@@ -0,0 +1,1409 @@
+	check_positions_rtl(sc, { position }, sep);
+}
+
+TEST_CASE("[SceneTree][SplitContainer] add children") {
```
I didn't know that test names should be capitalized, since TextEdit tests don't. I didn't do it for TabBar or TabContainer tests either, but I guess those can be fixed another time.

AThousandShips replied:
Generally they should, older tests are out of date, so best to fix for the new ones and keep the rest tracked for future fixes!

kitbdev commented:
Rebased and applied changes.
Some of the tests may seem pretty repetitive, but I couldn't find a better way that wouldn't sacrifice readability and I wanted coverage for each configuration of child expand flags for most things, since they can affect the internal behavior quite a bit.
Also added a case for 'Keep split offset when changing minimum size', for the behavior mentioned in the second part of https://github.com/godotengine/godot/pull/90411#issuecomment-2044406953.

akien-mga commented:
Thanks!
